.github/workflows/test.yaml

```diff
@@ -7,12 +7,12 @@ jobs:
   generate:
     strategy:
       matrix:
-        os: [ubuntu-latest, macos-latest, windows-latest]
+        os: [ubuntu-latest, macos-latest, windows-2019]
```

Comment:
CUDA v11 requires MSVC 2019, and `windows-latest` unfortunately only has MSVC 2022.
Before we let it slide, but it would only be able to build CPU mode, but I've refined the generate ps1 script to check for the expected MSVC version and error out with a friendly message if you have the wrong version wired up.  Without that, it fails in obscure ways later on in the build.

Reply:
That makes sense, generally building with older OSs and SDKs will yield better compatibility

Reply:
I've softened the requirement in the build scripts, but I want to leave it this way in CI since ultimately I want to inject CUDA and ROCm into the windows build and not just build CPU variants so we can build official builds with CI, and CUDA v11 requires 2019.